Improve error docs (#81)
earsketch/earsketch-curriculum: src/locales/en/v1/every-error-explained-in-detail.adoc

== Every Error Explained in Detail
:nofooter:

[[othererrors]]
=== Overview

> An error is a mistake in the code that prevents the program from running.

Reviewer: hm... not all errors prevent programs from running.

Author: In this sense, we're talking about `throw Error` though. Is it possible to throw an error without the program stopping?

Maybe I can re-word this to avoid the issue.

Author: From the curriculum 1.4
> Sometimes programmers make mistakes that cause code to work incorrectly, or not run at all. In programming, coding faults are called errors, or bugs. The process of finding and fixing bugs is called debugging. You can use debugging strategies, using the console.

Check the console for details and the line number. In some cases you may need to fix _the preceding line_ to resolve the error.

Guidance on addressing the most common errors is given in the next sections.

[role="curriculum-python"]
See the official https://docs.python.org/3/library/exceptions.html#concrete-exceptions[Python documentation^] for a full list of errors.

[role="curriculum-javascript"]

> See the official https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors[MDN documentation^] for a full list of errors in JavaScript.

Reviewer: nit: make the ending consistent with the python version; either drop "in JavaScript" here, or add "in Python" there.

Author: I think it makes sense to clarify "MDN documentation ... in JavaScript" - but not needed for "Python documentation ... in python". ¯\_(ツ)_/¯

Reviewer 2: ```suggestion
See Mozilla's https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors[JavaScript documentation^] for a full list of errors.
```

Reviewer 2: the "official" javascript docs are the ECMAscript language spec. MDN is just one third-party (albeit a very good one) resource. I don't think any students or teachers are going to know what MDN is, and they probably won't know what Mozilla is either.
